index.js

```diff
@@ -93,14 +93,14 @@ module.exports.sync = function(dir, iteratorSync, options){
   options = options || {};
   var initialDir = dir;
   var currentDepth = 0;
-  while(dir !== Path.join(dir, '..')){
+  while(dir !== Path.resolve(dir, '..')){
     if (typeof options.maxdepth === 'number' && options.maxdepth >= 0 && currentDepth > options.maxdepth) {
       break
     }
     currentDepth++;
     if(dir.indexOf('../../') !== -1 ) throw new Error(initialDir + ' is not correct.');
```

Comment:
This is the problematic line. I think it is intended to check against your use case. Once the root has been reached `path.join` adds `../` to the path which should be detected here and an (debateably  unhelpful) error thrown.

However on windows the path libary automatically converts paths to use `\\` instead of `/`. Therefore this check does not work as intended on windows.